src/ucs/sys/sys.c

```diff
@@ -1565,9 +1590,29 @@ ucs_status_t ucs_sys_get_memlock_rlimit(size_t *rlimit_value)
     *rlimit_value = (limit_info.rlim_cur == RLIM_INFINITY) ?
                             SIZE_MAX :
                             limit_info.rlim_cur;
+
     return UCS_OK;
 }
 
+ucs_status_t ucs_sys_get_effective_memlock_rlimit(size_t *rlimit_value)
+{
+    int mlock_cap = 0;
+
+    ucs_status_t status = ucs_sys_get_mlock_cap(&mlock_cap);
+    if (status != UCS_OK) {
+        return status;
```

Comment:
Do we really want to return "unsupported" from `ucs_sys_get_effective_memlock_rlimit`, just because the "capability API" isn't there?

Reply:
well, there are 3 options I see we can consider instead:
1) replace it with some other, more informative error code.
2) fallback to checking uid = 0.
3) fallback to using raw kernel API to query caps.

I think first option is best because:
- option 2 is not exactly what this function should do.
- option 3 is not portable between kernel versions and may break in the future.

Reply:
Why not simply ignore it and take the value from `ucs_sys_get_memlock_rlimit`?